core/frontend/src/LocalWebViewerApp.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module IModelApp
+ */
+
+import { IpcWebSocket, IpcWebSocketFrontend, IpcWebSocketMessage, IpcWebSocketTransport } from "@bentley/imodeljs-common";
+import { IpcApp } from "./IpcApp";
+
+class LocalTransport extends IpcWebSocketTransport {
+  private _client: WebSocket;
+
+  public constructor(port: number) {
+    super();
+
+    this._client = new WebSocket(`ws://localhost:${port}/`);
+
+    this._client.addEventListener("message", async (event) => {
+      for (const listener of IpcWebSocket.receivers)
+        listener({} as Event, JSON.parse(event.data as string));
+    });
+  }
+
+  public send(message: IpcWebSocketMessage): void {
+    this._client.send(JSON.stringify(message));
+  }
+}
+
+/** @alpha */
+export class LocalWebViewerApp {
```

Comment:
@swbsi I think this class should be called "LocalhostIpcApp" or something, if it is meant to be used only for apps with a localhost backend. Probably should be @internal rather than @alpha, since it is really only useful for test apps.

Reply:
How about LocalWebApp?
And change the function to initializeIpcForTesting?

Reply:
I prefer LocalhostIpcApp. It should be obvious that this is only useful for localhost and the reason you use it is so you can test Ipc. I think the "localhost" part tells people it's only for testing, and all "xxxApp" should have a "startup" method.

Reply:
Sounds good. I was just trying to keep "app" and "host" out of the same class name...

Reply:
yeah, i see that the *host* part of localhost can be confusing in this context. Less so when it's not capitalized. But I still think *localhost* is recognizable enough as one-word that it's the right thing here. I'll take the blame if anyone complains :).

Reply:
@swbsi see my changes so that you only call one `.startup` method.

But, i can't seem to get it to work. I don't see how my changes caused it, but i always get:
```
Error: listen EADDRINUSE: address already in use :::3002
```

Can you see if it works for you?

Reply:
I will give it a try.

Reply:
DTA works fine for me.